x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -70,6 +70,24 @@
                         new RoleDescriptor.IndicesPrivileges[] {
                                 RoleDescriptor.IndicesPrivileges.builder().indices(".monitoring-*").privileges("all").build() },
                         null, MetadataUtils.DEFAULT_RESERVED_METADATA))
+                .put("stack_monitoring_agent", new RoleDescriptor(
+                        "stack_monitoring_agent",
+                        new String[] {
+                            "monitor",
+                            "manage_index_templates"
+                        },
+                        new RoleDescriptor.IndicesPrivileges[] {
+                            RoleDescriptor.IndicesPrivileges.builder().indices("*").privileges("monitor").build(),
+                            RoleDescriptor.IndicesPrivileges.builder().indices(".kibana*").privileges("read").build(),
```

Comment:
Also, could you please explain why `.indices(".kibana*").privileges("read")` is required?

Reply:
I am not sorry, but I understand that Metricbeat calls a Kibana API and does not require to query the `.kibana` index (that API might do it on its account, but Metricbeat does not handle it itself). Please correct me here if I'm wrong.
The `.kibana` index is a treasure trove, we should be wary when reading it outside of through Kibana itself.

Reply:
It's correct that Beats does not query the index directly but through the API. As far as I understood it so far still the user needs to have these access rights so the API call is executed correctly. @ycombinator @chrisronline can perhaps confirm or correct?

Reply:
The user which metricbeat is using does need `.kibana` privileges as the user's role is used to determine permissions within the Kibana api. This particular api needs to query against the `.kibana` index so the user who makes the api request needs the appropriate permissions

Reply:
> The .kibana index is a treasure trove, we should be wary when reading it outside of through Kibana itself.

Adding to what @ruflin and @chrisronline said, the `GET /api/settings` Kibana API that Metricbeat calls reads the `.kibana` index, but only exposes a very narrow view of what's in that index. Here's the result of a call I just made locally while using the `elastic` superuser:

```js
{
   "cluster_uuid":"zLXBfdzPSrO12OBIz3ybyg",
   "settings":{
      "xpack":{
         "default_admin_email":"foo@bar.com"
      },
      "kibana":{
         "uuid":"5b2de169-2785-441b-ae8c-186a1936b17d",
         "name":"Shaunaks-MacBook-Pro-2.local",
         "index":".kibana",
         "host":"localhost",
         "transport_address":"localhost:5601",
         "version":"7.0.0-alpha1",
         "snapshot":false,
         "status":"green"
      }
   }
}
```

It may not be super obvious from the code in Kibana for this API, but it only returns the `xPack:defaultAdminEmail` setting and other metadata about the kibana instance:

https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/server/kibana_monitoring/collectors/get_settings_collector.js

Reply:
Okay understood! Thank you @ycombinator and @chrisronline .